lib/bosh/BoshDirectorClient.js

```diff
@@ -304,25 +304,40 @@ class BoshDirectorClient extends HttpClient {
   createOrUpdateDeployment(manifest, opts) {
     const query = opts ? _.pick(opts, 'recreate', 'skip_drain', 'context') : undefined;
     const deploymentName = yaml.safeLoad(manifest).name;
-    const config = _.sample(this.activePrimary);
-    if (config === undefined || config.length === 0) {
-      throw new errors.NotFound('Did not find any bosh director config which supports creation of deployment');
-    }
-    return this
-      .makeRequestWithConfig({
-        method: 'POST',
-        url: '/deployments',
-        headers: {
-          'Content-Type': 'text/yaml'
-        },
-        qs: query,
-        body: _.isObject(manifest) ? yaml.safeDump(manifest) : manifest
-      }, 302, config)
-      .tap(() => {
-        logger.info(`Cached ${deploymentName} at director: ${config.name} ${config.url}`);
-        this.cache[deploymentName] = config;
+    const boshDirectorName = _.get(opts, 'parameters.bosh_director_name');
+    return Promise.try(() => {
+        if (boshDirectorName) {
+          logger.info(`Deploying through ${boshDirectorName}`);
+          return this.getConfigByName(boshDirectorName);
```

Comment:
Just to be safe here, put the following  additional checks here:
1. `this.getDirectorConfig(deploymentName)` must null - if null, then return `this.getConfigByName(boshDirectorName)` else throw exception. In no scenario can someone call an update with passing director name.

Reply:
Done